### PR TITLE
feat: [SRM-13474]: implementing more scenarios for common Health source Design

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Proper loading message shall be shown while fetching the data
2. CSS refactoring for Records and Charts.
3. Update customMetricsMap to queryMetricsMap
4. On moving metrics to a different group name - it should not show a unique metric identifier.
5. Showing all the elements visible by default.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts [hunk 1]
```diff
@@ -8,12 +8,13 @@ export function shouldAutoBuildChart(
 }
 
 export function shouldShowChartComponent(
-  chartConfig: { enabled: boolean; chartVisibilityMode: CHART_VISIBILITY_ENUM } | undefined,
-  records: Record<string, any>[],
-  fetchingSampleRecordLoading: boolean,
-  query: string
+  chartConfig: { enabled: boolean; chartVisibilityMode: CHART_VISIBILITY_ENUM } | undefined
+  // records: Record<string, any>[],
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts:12]: This code might be useful later that'y its commented.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/__tests__/CommonCustomMetricFormContainer.test.tsx [hunk 2]
```diff
@@ -156,18 +156,18 @@ describe('Unit tests for CommonCustomMetricFormContainer', () => {
...
-    const chartConfig = { enabled: true, chartVisibilityMode: CHART_VISIBILITY_ENUM.AUTO }
-    const records = [] as Record<string, any>[]
-    expect(shouldShowChartComponent(chartConfig, records, false, 'query')).toEqual(false)
-  })
-  test('should not show Chart component when query is not present', async () => {
-    const chartConfig = { enabled: true, chartVisibilityMode: CHART_VISIBILITY_ENUM.AUTO }
-    const records = [{ record1: 'record-1' }]
-    expect(shouldShowChartComponent(chartConfig, records, false, '')).toEqual(false)
-  })
+  // test('should not show Chart component when records are not present and chart section is enabled in healthsource config', async () => {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/__tests__/CommonCustomMetricFormContainer.test.tsx:163]: same with the testcases. They might be useful later.

**Review by praneshg239 — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.utils.ts [hunk 3]
```diff
@@ -39,13 +42,41 @@ export const validateAddMetricForm = (
...
+  getString
+}: {
+  createdMetrics: string[]
+  metricName: string
+  groupedCreatedMetrics: GroupedCreatedMetrics
+  groupName: string | SelectOption
+  errors: FormikErrors<AddMetricForm>
+  getString: UseStringsReturn['getString']
+}): void {
   if (createdMetrics?.filter((name: string) => name === metricName).length) {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.utils.ts:65]: We should use .some()  here

Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/CustomMetricFormContainer.utils.ts:65]: This internally does the same thing, can pick it up in the next PR.

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**